treeherder/model/derived/jobs.py

```diff
@@ -388,17 +393,25 @@ def get_performance_series_summary(self, interval_seconds):
         # received data for the time interval requested
         last_updated_limit = utils.get_now_timestamp() - interval_seconds
 
-        data = self.get_jobs_dhub().execute(
-            proc="jobs.selects.get_perf_series_properties",
-            placeholders=[last_updated_limit, interval_seconds],
-            debug_show=self.DEBUG,
-        )
-        series_summary = defaultdict(lambda: {})
-        for datum in data:
-            (key, val) = (datum['property'], datum['value'])
-            if key == 'subtest_signatures':
-                val = json.loads(val)
-            series_summary[datum['signature']][key] = val
+        cache_key = self._get_performance_series_cache_key(interval_seconds)
+        series_summary = cache.get(cache_key, None)
+        if series_summary:
+            series_summary = json.loads(series_summary)
+        else:
+            data = self.get_jobs_dhub().execute(
+                proc="jobs.selects.get_perf_series_properties",
+                placeholders=[last_updated_limit, interval_seconds],
+                debug_show=self.DEBUG,
+            )
+
+            series_summary = defaultdict(lambda: {})
+            for datum in data:
+                (key, val) = (datum['property'], datum['value'])
+                if key == 'subtest_signatures':
+                    val = json.loads(val)
+                series_summary[datum['signature']][key] = val
+
+            cache.set(cache_key, json.dumps(series_summary))
```

Comment:
You're right, though we do need to convert it to type dict() instead of defaultdict first. Since we're doing the zlib hack though, we'll still need to do this.